postprocessing/janus-pp-rec.c

```diff
@@ -1114,6 +1156,57 @@ int main(int argc, char *argv[])
 		exit(1);
 	}
 
+	/* Run restamping */
+	if(!video && !data && restamp_multiplier > 0) {
+		tmp = list;
+		uint64_t restamping_offset = 0;
+		double restamp_threshold = (double) restamp_min_th/1000;
+		double restamp_jump_multiplier = (double) restamp_multiplier/1000;
+
+		while(tmp) {
+			uint64_t original_ts = tmp->ts;
+
+			/* Update ts with current offset */
+			if(restamping_offset > 0) {
+				tmp->ts += restamping_offset;
+			}
+
+			/* Calculate diff between time of reception and the rtp timestamp */
+			double current_latency = get_latency(tmp, rate);
+
+			if(current_latency > restamp_threshold) {
+				/* Check for possible jump compared to previous latency values */
+				double moving_avg_latency = get_moving_average_of_latency(tmp->prev, rate, restamp_packets);
+				JANUS_LOG(LOG_VERB, "latency=%.2f mavg=%.2f\n", current_latency, moving_avg_latency);
+
+				/* Found a new jump in latency? */
+				if(current_latency > (moving_avg_latency*restamp_jump_multiplier)) {
+					/* Increase restamping offset with current offset */
+					restamping_offset += (current_latency-moving_avg_latency)*rate;
+
+					/* Calculate new_ts with new offset */
+					uint64_t new_ts = original_ts+restamping_offset;
+
+					/* Checking new_ts against previous ts to avoid converting with wrong timestamps */
+					if(new_ts < tmp->prev->ts) {
+						/* Avoid creating unnecessary large files... */
+						JANUS_LOG(LOG_ERR, "new ts would be smaller than previous ts! new_ts=%.ld prev_ts=%.ld\n", new_ts, tmp->prev->ts);
+						exit(1);
+					}
+
+					/* Update current packet ts with new ts */
+					tmp->ts = new_ts;
+
+					JANUS_LOG(LOG_WARN, "Gap detected. Restamping packets from here. Seq: %d\n", tmp->seq);
```

Comment:
What about the sequence numbers ? Why they are not being compensated ?

Reply:
Since the sequence numbers are correct only the timestamps are wrong, I did not think the sequence numbers should be compensated.

Reply:
You are right.
Anyway I was thinking that changing the log line to "**Timestamp** gap detected" would be more accurate.
Gap sounds like "packet loss" to me.

Reply:
I updated the wording. Your version makes more sense. :+1: